### DISCUSSION TITLE
Problem when using Local Model and chat with note

### DISCUSSION THREAD

**Discussion by haluwong** [logancyang/obsidian-copilot#539] [Q&A]
Running Ollama with llama2:13b

When using "Chat" mode to ask question about the active note. It will not working as expected.

Attached with the screen capture.

Is it the problem in Ollama side or related the input context length?

![screen_20240322_09](https://github.com/logancyang/obsidian-copilot/assets/2929231/bc473640-65c6-4328-b5bc-2535ba2c4fee)

[INSTALL.md](https://github.com/logancyang/obsidian-copilot/files/14721286/INSTALL.md)

**Comment by logancyang** [comment]
It seems the note was passed correctly. You can click BufferWindowMemory and find the messages and confirm it the previous note is in it. It should be, given the screenshot.

My biggest suspicion is your context window setting in Ollama. Can you refer to https://github.com/logancyang/obsidian-copilot/blob/master/local_copilot.md#ollama and make sure you set the context window correctly. I recommend Mistral models with a 32k window if your machine has enough RAM.
